charms/jupyter-controller/tests/integration/test_charm.py

```diff
@@ -161,3 +164,38 @@ async def test_prometheus_integration(ops_test: OpsTest):
     wait=tenacity.wait_exponential(multiplier=1, min=1, max=10),
     reraise=True,
 )
+
+
+@pytest.mark.abort_on_fail
+async def test_remove_with_resources_present(ops_test: OpsTest):
+    """Test remove with all resources deployed.
+
+    Verify that all deployed resources that need to be removed are removed.
+    """
+
+    # remove deployed charm and verify that it is removed
+    await ops_test.model.remove_application(app_name=APP_NAME, block_until_done=True)
+    assert APP_NAME not in ops_test.model.applications
+
+    # verify that all resources that were deployed are removed
+    lightkube_client = Client()
+
+    # verify all CRDs in namespace are removed
+    crd_list = lightkube_client.list(
+        CustomResourceDefinition,
+        labels=[("app.juju.is/created-by", "jupyter-controller")],
+        namespace=ops_test.model.name,
+    )
```

Comment:
Do things we've created via lightkube get these labels?  I'm thinking these might be a podspec thing, but may be remembering wrong

Reply:
That's the labels from the actual deployment. So, whatever we create in charm with lightkube has this label.